platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java

```diff
@@ -213,13 +215,15 @@ private void addDefaultReportAndCoverageVerificationTasks(final JacocoPluginExte
             JvmTestSuite defaultTestSuite = testing.getSuites().withType(JvmTestSuite.class).getByName(JvmTestSuitePlugin.DEFAULT_TEST_SUITE_NAME);
             defaultTestSuite.getTargets().configureEach(target -> {
                 TaskProvider<Test> testTask = target.getTestTask();
-                addDefaultReportTask(extension, testTask);
-                addDefaultCoverageVerificationTask(testTask);
+                SourceSet mainSourceSet = project.getExtensions().getByType(SourceSetContainer.class).getByName("main");
```

Comment:
I think that this pr is good in that it adds the _option_ to configure the source encoding for jacoco, but I don't necessarily think we should link the default value to the compile task. 

As per our previous comments, there really should be a higher-level property on the source sets that sets this value for the compile task, jacoco task, etc.

I realize that we do not need to add that wiring now, but I'm not we necessarily want this wiring here in the meantime

Reply:
In the current situation where there is no such central setting yet, what would be a reason not to want this wiring?
The `compileJava` task defines which encoding to use to read the source files, the JaCoCo task is also reading the source files, so should use the same encoding.
It might be correct, that there could be other compile tasks that are configured differently.
But I thought this would at least be consistent, for example with `TARGET_JVM_VERSION_ATTRIBUTE` default value being taken from `compileJava` too.

Reply:
A higher-level property would be a really good solution. But for the moment, I prefer to have this additional property (like all the other additional properties on other tasks), than having to wait that someone implements the higher-level property. This pr is a solution for an existing problem - an overall higher-level solutions should be separated into a new ticket, a new pr and new planning.

Reply:
We discussed this, and we have decided that we should introduce a new property on the jacoco task but not add the link to the java compile task. 

The reason for this is that if we add the convention now, future changes to the convention will be a breaking change. For example consider a case where the convention as it exists in this PR is merged. Now, imagine the user sets the encoding on the compile task, the jacoco task will get that value by convention. When we add a higher-level property to set the convention for both the compile task and the jacoco report task, and the user continues to only set the compile task value, the jacoco report task will lose its prior conventional value as it no longer gets its conventional value from the compile task but from the higher level property. 

If you remove the convention link between the compile task and the jacoco task we can get this in for 8.8.

Reply:
As you wish, connection is cut. :-)